### ISSUE TITLE
Error building ZMQ on Windows

### ISSUE THREAD

**Issue by AndyGreenwell** [JuliaLang/ZMQ.jl#90]
While attempting to install ZMQ.jl as part of an IJulia installation for a 32-bit Julia 0.3.11 on a 64-bit Windows Server 2012 environment (I have justification for such madness), I am receiving build errors for both Nettle and ZMQ. I have filed a similar [issue](https://github.com/staticfloat/Nettle.jl/issues/48) under Nettle.jl.

Below is the terminal output when attempting to do a Pkg.add("IJulia") as part of a new Julia installation on a new OS installation:

----------------------------------------------------------------
```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "help()" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.3.11 (2015-07-27 06:18 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  i686-w64-mingw32

julia> Pkg.add("IJulia")
INFO: Initializing package repository C:\Users\Administrator\.julia\v0.3
INFO: Cloning METADATA from git://github.com/JuliaLang/METADATA.jl
INFO: Cloning cache of BinDeps from git://github.com/JuliaLang/BinDeps.jl.git
INFO: Cloning cache of Compat from git://github.com/JuliaLang/Compat.jl.git
INFO: Cloning cache of IJulia from git://github.com/JuliaLang/IJulia.jl.git
INFO: Cloning cache of JSON from git://github.com/JuliaLang/JSON.jl.git
INFO: Cloning cache of LibExpat from git://github.com/amitmurthy/LibExpat.jl.git

INFO: Cloning cache of Nettle from git://github.com/staticfloat/Nettle.jl.git
INFO: Cloning cache of REPLCompletions from git://github.com/Keno/REPLCompletion
s.jl.git
INFO: Cloning cache of SHA from git://github.com/staticfloat/SHA.jl.git
INFO: Cloning cache of URIParser from git://github.com/JuliaWeb/URIParser.jl.git

INFO: Cloning cache of WinRPM from git://github.com/JuliaLang/WinRPM.jl.git
INFO: Cloning cache of ZMQ from git://github.com/JuliaLang/ZMQ.jl.git
INFO: Cloning cache of Zlib from git://github.com/dcjones/Zlib.jl.git
INFO: Installing BinDeps v0.3.15
INFO: Installing Compat v0.6.0
INFO: Installing IJulia v0.2.5
INFO: Installing JSON v0.4.5
INFO: Installing LibExpat v0.0.8
INFO: Installing Nettle v0.1.10
INFO: Installing REPLCompletions v0.0.3
INFO: Installing SHA v0.1.1
INFO: Installing URIParser v0.0.5
INFO: Installing WinRPM v0.1.12
INFO: Installing ZMQ v0.2.0
INFO: Installing Zlib v0.1.8
INFO: Building WinRPM
WARNING: skipping repodata/repomd.xml, not in cache -- call WinRPM.update() to d
ownload
WARNING: skipping repodata/repomd.xml, not in cache -- call WinRPM.update() to d
ownload
INFO: Downloading https://cache.e.ip.saba.us/http://download.opensuse.org/reposi
tories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 1/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 2/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 3/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 4/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 5/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: received error 0 while downloading https://cache.e.ip.saba.us/http://do
wnload.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/re
pomd.xml
INFO: Downloading https://cache.e.ip.saba.us/http://download.opensuse.org/reposi
tories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 1/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 2/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 3/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 4/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 5/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: received error 0 while downloading https://cache.e.ip.saba.us/http://do
wnload.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/re
pomd.xml
INFO: Building Nettle
INFO: Updating WinRPM package list
INFO: Downloading https://cache.e.ip.saba.us/http://download.opensuse.org/reposi
tories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 1/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 2/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 3/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 4/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 5/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/repomd.xml
WARNING: received error 0 while downloading https://cache.e.ip.saba.us/http://do
wnload.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_13.1/repodata/re
pomd.xml
INFO: Downloading https://cache.e.ip.saba.us/http://download.opensuse.org/reposi
tories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 1/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 2/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 3/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 4/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: Unknown download failure, error code: 2148270094
WARNING: Retry 5/5 downloading: https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
WARNING: received error 0 while downloading https://cache.e.ip.saba.us/http://do
wnload.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/re
pomd.xml
===============================[ ERROR: Nettle ]================================


None of the selected providers can install dependency nettle.
Use BinDeps.debug(package_name) to see available providers

while loading C:\Users\Administrator\.julia\v0.3\Nettle\deps\build.jl, in expres
sion starting on line 39

================================================================================

INFO: Building ZMQ
=================================[ ERROR: ZMQ ]=================================


None of the selected providers can install dependency zmq.
Use BinDeps.debug(package_name) to see available providers

while loading C:\Users\Administrator\.julia\v0.3\ZMQ\deps\build.jl, in expressio
n starting on line 23

================================================================================

INFO: Building IJulia
Found IPython version 3.2.0 ... ok.
Creating julia profile in IPython...
Changing KernelManager.kernel_cmd to ["C:\\\\Users\\\\Administrator\\\\AppData\\
\\Local\\\\Julia-0.3.11\\\\bin\\\\julia.exe","-i","-F","C:\\\\Users\\\\Administr
ator\\\\.julia\\\\v0.3\\\\IJulia\\\\src\\\\kernel.jl","{connection_file}"] in ip
ython_config.py...
(Existing IPythonWidget.execute_on_complete_input setting in ipython_qtconsole_c
onfig.py is untouched.)
(Existing FrontendWidget.lexer_class setting in ipython_qtconsole_config.py is u
ntouched.)
(Existing NotebookApp.port setting in ipython_notebook_config.py is untouched.)
(Existing ipynblogo.png file untouched.)
(Existing ipynblogo.svg file untouched.)
(Existing favicon.ico file untouched.)
(Existing custom.js file untouched.)
(Existing julia.js file untouched.)
Writing IJulia kernelspec to C:\Users\Administrator\.ipython\kernels\julia-0.3\k
ernel.json ...
(Existing logo-32x32.png file untouched.)
(Existing logo-64x64.png file untouched.)

================================[ BUILD ERRORS ]================================


WARNING: Nettle and ZMQ had build errors.

 - packages with build errors remain installed in C:\Users\Administrator\.julia\
v0.3
 - build the package(s) and all dependencies with `Pkg.build("Nettle", "ZMQ")`
 - build a single package by running its `deps/build.jl` script

================================================================================

INFO: Package database updated
```

**Comment by yuyichao** [comment]
If I read correctly this is a download error?

**Comment by yuyichao** [comment]
Ah, I see the download issue is already pointed out in the other issue. @tkelman as windows expert and I remember there's some other existing issue about downloading, especially on windows.

**Comment by vtjnash** [comment]
maybe an issue with https://github.com/staticfloat/cache.julialang.org?

`0x800C000E : Navigation request has encountered a security issue.`
that's not very helpful. does `Base.download("https://cache.e.ip.saba.us/http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml"` or `Base.download("http://download.opens
use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml")` work?

**Comment by vtjnash** [comment]
edit: does IE consider the certificate of https://cache.e.ip.saba.us/ to be valid?

**Comment by AndyGreenwell** [comment]
As the settings for IE in my OS image are at their defaults, IE Enhanced Security Configuration is configured to On for my current user.  The IE session did not put up a warning when browsing to https://cache.e.ip.saba.us and I am attaching a screenshot of both the webpage and the associated certificate info viewable from the browser.

![certificatescreenshot](https://cloud.githubusercontent.com/assets/11096243/9450358/29b6209e-4a76-11e5-9b39-31b7b3b6912f.PNG)

**Comment by AndyGreenwell** [comment]
The Base.download commands do not work for me.

```
julia> Base.download("https://cache.e.ip.saba.us/http://download.opens use.org/r
epositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml")
ERROR: automatic download failed (error: 2148270094): https://cache.e.ip.saba.us
/http://download.opens use.org/repositories/windows:/mingw:/win64/openSUSE_13.1/
repodata/repomd.xml
 in error at error.jl:21
 in download at interactiveutil.jl:329
 in download at interactiveutil.jl:336

julia> Base.download("http://download.opens use.org/repositories/windows:/mingw:
/win64/openSUSE_13.1/repodata/repomd.xml")
ERROR: automatic download failed (error: 2148270085): http://download.opens use.
org/repositories/windows:/mingw:/win64/openSUSE_13.1/repodata/repomd.xml
 in error at error.jl:21
 in download at interactiveutil.jl:329
 in download at interactiveutil.jl:336
```

**Comment by AndyGreenwell** [comment]
Closing as this problem seemed to be related to the IE Enhanced Security Configuration setting being set to on, so this was essentially a firewall issue.

**Comment by tkelman** [comment]
This is probably a duplicate of https://github.com/JuliaLang/WinRPM.jl/issues/47 - maybe we could turn the caching back off if the certificates are giving people problems (@staticfloat anything we could do about that?), but without the caching the opensuse mirrors would occasionally be down in an annoyingly geographically-dependent way.

**Comment by staticfloat** [comment]
I don't think there is anything wrong with the certificates; this is something special that is induced by Windows Server wanting to keep as tight a lock as possible on internet connections.  @AndyGreenwell am I correct in thinking that you've sorted out this problem?  If possible, I'd like to get things working for you before closing the issue.

**Comment by tkelman** [comment]
(let's move this to https://github.com/JuliaLang/WinRPM.jl/issues/47)

**Comment by AndyGreenwell** [comment]
Whenever I was able to turn off Internet Explorer Enhanced Security Configuration appropriately, I did not have an issue with downloading these files, and then installing and building Nettle.jl and ZMQ.jl on my machine.  

I just spent a bit of time building up a server on AWS to install this configuration, and for the Administrator account, once the IE ESC was disabled, the installation went smoothly for me.  On non-Administrator accounts, or for accounts that I created later and then gave Administrative privileges, the IE ESC setting was still acting as though it was enabled, even after I disabled the setting in the Server Manager and in the Registry.  Consequently, for those other accounts on that server that were not named "Administrator", I was not able to install either Nettle.jl or ZMQ.jl.

I will work on performing the tests that @tkelman suggests in https://github.com/JuliaLang/WinRPM.jl/issues/47
